### ISSUE TITLE
[SPEC] Enterprise AI Compliance Scoring — EU AI Act, NIST AI RMF & Supply Chain Trust Framework

### ISSUE THREAD

**Issue by croissantsam** [smcroissant/llmtrust#88]
# [SPEC] Enterprise AI Compliance Scoring

> Covers: #81 [STRATEGIC] Trust Signals for Enterprise AI Procurement
> Author: David (PM) — CroissantLabs 🥐
> Date: 2026-03-16

---

## 1. Executive Summary

Enterprise AI procurement is exploding but trust evaluation is fragmented and manual. CISOs and procurement teams need **auditable compliance scores** before approving LLM deployments. Nobody currently scores LLMs against regulatory frameworks (EU AI Act, NIST AI RMF) or supply chain security standards.

LLMTrust can own the **enterprise compliance layer** — the thing procurement teams check before signing contracts. This spec defines the compliance scoring framework, data sources, and API design.

---

### Regulatory Tailwinds
- **EU AI Act** (effective 2025-2026) — Mandatory risk classification for AI systems. Enterprises deploying LLMs must classify their use case (minimal/high/unacceptable risk) and document compliance.
- **NIST AI RMF 1.0** — Voluntary framework widely adopted by US enterprises. Provides structured risk categories (governance, mapping, measuring, managing).
- **SOC 2 / ISO 27001** — Enterprise procurement baseline. LLM providers increasingly need to demonstrate alignment.
- **GDPR** — Data provenance for training data matters for EU deployments.

### Competitive Gap
| Platform | Performance Scoring | Compliance Scoring |
|----------|-------------------|-------------------|
| Artificial Analysis | ✅ | ❌ |
| HuggingFace | ✅ | ❌ |
| Arena.ai | ✅ | ❌ |
| OpenRouter | ✅ | ❌ |
| **LLMTrust** | ✅ (via #84) | **✅ (this spec)** |

**Nobody scores:** license compliance, supply chain security, data provenance, bias/fairness metrics, regulatory readiness.

---

#### A. Regulatory Alignment (30% weight)
| Check | Framework | Scoring Method |
|-------|-----------|---------------|
| Risk tier classification | EU AI Act | Model purpose analysis + automated checks |
| Transparency requirements | EU AI Act Art. 52 | Model card completeness, watermarking |
| Human oversight readiness | EU AI Act Art. 14 | API-level controls, logging |
| Risk management process | NIST AI RMF GOVERN-1 | Documentation scoring |
| Bias testing | NIST AI RMF MEASURE-1 | Automated benchmark suite |

#### B. Supply Chain Security (25% weight)
| Check | Standard | Scoring Method |
|-------|----------|---------------|
| Model provenance | SLSA framework | HuggingFace metadata, checksum verification |
| Training data disclosure | Data Cards | Model card analysis |
| Dependency audit | Custom | Fine-tuned model lineage |
| License clarity | SPDX | License field analysis (permissive vs restrictive) |
| Vulnerability disclosure | CVE pattern | Security advisory history |

#### C. Data Governance (20% weight)
| Check | Standard | Scoring Method |
|-------|----------|---------------|
| Training data consent | GDPR/CCPA | Opt-out compliance, data sourcing |
| PII handling | GDPR Art. 5 | Data minimization documentation |
| Data retention policy | SOC 2 CC6.1 | Provider policy analysis |
| Cross-border transfer | GDPR Ch. V | Data residency documentation |

#### D. Operational Trust (15% weight)
| Check | Standard | Scoring Method |
|-------|----------|---------------|
| SLA availability | Provider SLA | Uptime guarantee analysis |
| Incident response | SOC 2 CC7.3 | Security page/documentation review |
| Penetration testing | SOC 2 CC7.1 | Security audit history |
| Business continuity | ISO 22301 | DR documentation |

#### E. Ethical AI (10% weight)
| Check | Framework | Scoring Method |
|-------|-----------|---------------|
| Bias fairness metrics | Algorithmic bias benchmarks | Standardized test suite |
| Harmful content filtering | Content policy analysis | Safety evaluation results |
| Environmental impact | Carbon footprint disclosure | Training compute estimates |

### 3.2 Composite Compliance Score
```
ComplianceScore = Σ(category_score × category_weight)

Category scores: Automated checks (60%) + Manual review (40% for Enterprise tier)
Score range: 0-100
```

### 3.3 Compliance Badges
| Badge | Threshold | Meaning |
|-------|-----------|---------|
| 🛡️ **Enterprise Ready** | ≥85 overall, no category <70 | Approved for regulated enterprise use |
| ✅ **Compliant** | ≥70 overall | Meets baseline compliance requirements |
| ⚠️ **Review Required** | 50-69 | Gaps identified, manual review needed |
| ❌ **Non-Compliant** | <50 | Significant compliance gaps |

---

### 4.1 Database Schema
```sql
-- Compliance checks (automated)
CREATE TABLE compliance_checks (
  id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
  model_id UUID REFERENCES models(id),
  provider TEXT NOT NULL,
  category TEXT NOT NULL,          -- "regulatory", "supply_chain", "data_governance", "operational", "ethical"
  check_name TEXT NOT NULL,
  check_result JSONB NOT NULL,     -- {score, evidence, confidence, details}
  check_type TEXT NOT NULL,        -- "automated", "manual"
  computed_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
  UNIQUE(model_id, category, check_name)
);

-- Composite compliance scores
CREATE TABLE compliance_scores (
  id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
  model_id UUID REFERENCES models(id),
  provider TEXT NOT NULL,
  overall_score REAL NOT NULL,
  category_scores JSONB NOT NULL,  -- {regulatory: 75, supply_chain: 82, ...}
  badge TEXT NOT NULL,             -- "enterprise_ready", "compliant", "review_required", "non_compliant"
  data_freshness TIMESTAMPTZ NOT NULL,
  computed_at TIMESTAMPTZ NOT NULL DEFAULT NOW(),
  UNIQUE(model_id)
);

-- Compliance reports (PDF export)
CREATE TABLE compliance_reports (
  id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
  model_id UUID REFERENCES models(id),
  report_type TEXT NOT NULL,       -- "eu_ai_act", "nist_ai_rmf", "soc2", "full"
  generated_by UUID REFERENCES users(id),
  report_url TEXT,                 -- S3/R2 URL for PDF
  generated_at TIMESTAMPTZ NOT NULL DEFAULT NOW()
);
```

### 4.2 Automated Check Pipeline
1. **Model Card Analyzer** — NLP extraction from HuggingFace model cards (license, data, limitations)
2. **Metadata Verifier** — Checksum validation, SLSA provenance verification
3. **License Classifier** — SPDX parsing + permissiveness scoring
4. **Safety Benchmark Runner** — Standardized bias/fairness test suite (weekly batch)
5. **Provider Policy Scraper** — Monitor provider security pages, SLA pages, ToS changes

### 4.3 Manual Review Process (Enterprise Tier)
- Enterprise customers can request manual compliance review
- Reviewers fill structured rubric aligned to framework categories
- Results feed into compliance score with higher weight
- Process: Request → Assign reviewer → Review (2-3 days) → Score update → PDF report

---

### Endpoints
```
GET  /v1/compliance-scores?model=X&provider=Y         — Current compliance score + breakdown
GET  /v1/compliance-scores/{id}/report?format=pdf      — Download compliance report
GET  /v1/compliance/frameworks                         — List supported frameworks (EU AI Act, NIST, SOC 2)
POST /v1/compliance/review-request                     — Request manual review (Enterprise tier)
GET  /v1/compliance/leaderboard?category=regulatory    — Ranked compliance scores
```

### Pricing Integration
- **Free tier:** Overall compliance badge only (no breakdown)
- **Pro (/mo):** Category breakdown + automated checks
- **Enterprise (custom):** Full reports, manual review, PDF export, API access

---

### PDF Report Structure
1. **Executive Summary** — Overall score, badge, key findings
2. **Regulatory Alignment** — EU AI Act tier, NIST alignment, gaps
3. **Supply Chain Security** — Provenance, license, dependencies
4. **Data Governance** — GDPR, data retention, PII handling
5. **Operational Trust** — SLA, incident response, security audits
6. **Ethical AI** — Bias metrics, safety evaluation
7. **Recommendations** — Actionable items to improve score
8. **Appendix** — Raw check results, evidence links

---

### Phase 1: Automated Checks (Month 1)
- Model card analysis, license verification, metadata checks
- 50 models scored with automated compliance
- Public compliance badges on model pages

### Phase 2: Framework Integration (Month 2)
- EU AI Act risk tier classification
- NIST AI RMF category mapping
- Compliance leaderboard + API

### Phase 3: Enterprise Sales (Month 3)
- Manual review process
- PDF report generation
- Enterprise pilot partnerships (2-3 target customers)

---

## 8. Success Metrics

| Metric | Target (30d) | Target (90d) |
|--------|-------------|-------------|
| Models with compliance scores | 50 | 200 |
| Automated checks per model | 25+ | 35+ |
| Enterprise pilot requests | 3 | 10 |
| Compliance API calls/day | 50 | 500 |
| PDF reports generated | — | 20 |

---

## 9. Dependencies

- **#59** [SPEC] Trust-as-a-Service API — API architecture foundation
- **#67** [FEATURE] Trust-as-a-Service API — delivery channel
- **#79** [Enterprise Tier] — pricing and access model
- **#66** [STRATEGIC] Credibility & Partnership Playbook — enterprise trust

---

## 10. Risks

| Risk | Impact | Mitigation |
|------|--------|------------|
| Regulatory frameworks evolve | Scoring criteria become outdated | Quarterly framework review process |
| False compliance claims | Enterprise trust erosion | Evidence-based scoring, cite sources |
| Manual review bottleneck | Slow enterprise onboarding | Start with automated; hire reviewers as demand grows |
| Provider resistance to data sharing | Incomplete supply chain data | Multiple sources (HF, direct scraping, community reports) |

---

## 11. Open Questions

1. Should we pursue formal partnership/accreditation with EU AI Act compliance bodies?
2. How to handle models with no training data disclosure (score 0 or "insufficient data")?
3. Should compliance scores be cached aggressively (weekly) or computed on-demand?
4. Do we need a legal review of our compliance scoring methodology?

---

*— David (PM), CroissantLabs 🥐*
